Add commitment burndown columns (#400)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/glossary.md

<a name="glossary:commitment-based-discount"><b>Commitment-Based Discount</b></a>

> A billing discount model that offers reduced rates on preselected SKUs in exchange for an obligated usage or spend amount over a predefined term.  Commitment-based discount purchases, made upfront and/or with recurring monthly payments are amortized evenly across predefined charge periods (i.e. hourly), and unused amounts cannot be carried over to subsequent charge periods.

Reviewer: ```suggestion
A billing discount model that offers reduced rates on preselected SKUs in exchange for an obligated usage or spend amount over a predefined term.  Commitment-based discount purchases, made upfront and/or with recurring monthly payments are amortized evenly across predefined charge periods (i.e. hourly), and unused amounts cannot be carried over to subsequent charge periods. Commitment discounts are publicly available to customers without special contract arrangements.
```

By adding this sentence (reference: [FinOps Terminology](https://www.finops.org/assets/terminology/#commitment-discounts)), we can exclude those 'problematic discounts' (e.g., OCI Universal Credits) that I initially classified as Commitment-Based Discounts because they match all other criteria we’ve discussed (e.g. concept of burn-down commitment, purchase amortization, etc.).

> A billing discount model that offers reduced rates on preselected SKUs in exchange for an obligated usage or spend amount over a predefined term.  Commitment-based discount purchases, made upfront and/or with recurring monthly payments are amortized evenly across predefined charge periods (i.e. hourly), and unused amounts cannot be carried over to subsequent charge periods. Commitment discounts are publicly available to customers without special contract arrangements.

Reviewer 2: have we defined amortization in the glossary? Effectivecost is the amortization amount but doesnt speak to amortization. If we haven't defined amortization in the glossary then would it be better to say 'payments are distributed evenly across'

Author: Matches this definition exactly, so leaving as is: https://www.finops.org/assets/terminology/#commitment-discounts

A Date/Time Format value that is contained within the beginning bound of a time period.

<a name="glossary:instance-family"><b>Instance Family</b></a>

A SKU property that groups compute-based SKUs with similar hardware and features that differ by allocated amounts of compute, memory, or storage.

<a name="glossary:instance-size"><b>Instance Size</b></a>

A SKU property denoting the SKU's relative computing capability within its [*instance family*](#glossary:instance-family).

> <a name="glossary:instance-type-ratio"><b>Instance Type Ratio</b></a>

Reviewer 3: This seems to be miss-named.

Reviewer 3: Also this is implying that the only supported approach to normalization factor is instance family and size. Normalization likely needs to be more generic to support multple normalization factors.

Reviewer 3: I aggree with @ijurica this needs more consensus before introduction. This is especially true if we are going to use it to introduce a normalization factor.

A term that describes a service that is available and provided immediately or as needed, without requiring a pre-scheduled appointment or prior arrangement. In cloud computing, virtual machines can be created and terminated as needed, i.e. on demand.

> <a name="glossary:negotiated-discount"><b>Negotiated Discount</b></a>

Reviewer 4: We should differentiate in Commitment discount related columns that it is not meant for the negotiated discounts (this is what we do for consumed vs pricing units - we explicitly state that one is for pricing and cost and the other is for consumption). Similar differentiation in commitment discount columns using this glossary entry will be useful.

Author: Applied ~"... confused with negotiated discounts ..." across all `CommitmentDiscount*` columns

> <a name="glossary:negotiated-discount"><b>Negotiated Discount</b></a>

Author: We've had this debate and this aligns with the negotiated discount definition added to the FinOps terminology page: https://www.finops.org/assets/terminology/#negotiated-discount

> A contractual agreement where a customer commits to specific spend or usage goals over a [*term*](#glossary:term) in exchange for discounted rates across varying SKUs.  Unlike [*commitment discounts*](#glossary:commitment-discount), negotiated discounts are typically more customized to customer's accounts, can be burned down at varying frequencies, and may overlap with *commitment discounts*.

Reviewer: ```suggestion
A contractual agreement where a customer commits to specific spend or usage goals over a [*term*](#glossary:term) in exchange for discounted rates across varying SKUs.  Unlike [*commitment discounts*](#glossary:commitment-discount), negotiated discounts are typically more customized to customer's accounts, can be utilized at varying frequencies, and may overlap with *commitment discounts*.
```

**utilized** is used in [FinOps terminology](https://www.finops.org/assets/terminology/) definition and covers both build-up and burn-down use cases, whereas **burned down** excludes build-up negotiated discounts - EDP, MACC, etc.

> A contractual agreement where a customer commits to specific spend or usage goals over a [*term*](#glossary:term) in exchange for discounted rates across varying SKUs.  Unlike [*commitment-based discounts*](#glossary:commitment-based-discount), negotiated discounts are typically more customized to customer's accounts, can be burned down at varying frequencies, and may be overlap with *commitment-based discounts*.

Reviewer 5: > Unlike *commitment-based discounts*  ... and may be overlap with *commitment-based discounts*.

This might raise questions for readers. Maybe a separate sentence about commitment-based discounts could clarify, like:

> Negotiated discounts are typically more customized to customer's accounts, can be burned down at varying frequencies. Negotiated discounts may apply to commitment-based discounts.

Author: This is the exact FinOps terminology (literally copy and pasted), so going to keep consistent

> A contractual agreement where a customer commits to specific spend or usage goals over a [*term*](#glossary:term) in exchange for discounted rates across varying SKUs.  Unlike [*commitment-based discounts*](#glossary:commitment-based-discount), negotiated discounts are typically more customized to customer's accounts, can be burned down at varying frequencies, and may be overlap with *commitment-based discounts*.

Reviewer 2: ```suggestion
A contractual agreement where a customer commits to specific spend or usage goals over a [*term*](#glossary:term) in exchange for discounted rates across varying SKUs.  Unlike [*commitment-based discounts*](#glossary:commitment-based-discount), negotiated discounts are typically more customized to customer's accounts, can be burned down at varying frequencies, and may overlap with *commitment-based discounts*.
```

<a name="glossary:size-flexibility"><b>Size Flexibility</b></a>

> A feature of [*commitment discounts*](#glossary:commitment-discount) that may transform the number of resource hours purchased or consumed into an amount based on additional, provider-specific, [*SKU*](glossary:sku) aspects.

Author: Use "Commitment Flexibility" and general as above
